File: 프로그래머스/2/12914. 멀리 뛰기/멀리 뛰기.py


def solution(n):
    answer = 0
    
    if n < 3:
        return n
    else:
        d = [0] * (n+1)
        d[0] = 1
        d[1] = 2
        d[2] = 0
        
        for i in range(3, n+1):
            d[2] = d[1] + d[0]
            d[0] = d[1]
            d[1] = d[2]
        return d[2] % 1234567

File: 프로그래머스/2/12914. 멀리 뛰기/멀리 뛰기.java
class Solution {
    public long solution(int n) {
        if(n < 3) {
            return n;
        }
        long[] jump = new long[3];
        jump[0] = 1;
        jump[1] = 2;
        int trial = 3;
        while (trial <= n){
            jump[2] = (jump[1] + jump[0])%1234567;
            jump[0] = jump[1];
            jump[1] = jump[2];
            trial++;
        }

        return jump[2];
    }
}
